git_machete/client.py

```diff
@@ -1512,39 +1512,55 @@ def __pick_remote(self, branch: str, is_called_from_traverse: bool) -> None:
+                   "repository, or 'q' to quit creating pull request: "
+
         ans = input(msg).lower()
         if ans in ('q', 'quit'):
-            raise StopTraversal
+            raise StopInteraction
         try:
             index = int(ans) - 1
             if index not in range(len(rems)):
                 raise MacheteException(f"Invalid index: {index + 1}")
             self.handle_untracked_branch(rems[index], branch, is_called_from_traverse)
         except ValueError:
-            pass
+            if not is_called_from_traverse:
+                raise MacheteException('Could not establish remote repository, pull request creation interrupted.')
 
     def handle_untracked_branch(self, new_remote: str, branch: str, is_called_from_traverse: bool) -> None:
         rems: List[str] = self.__git.remotes()
         can_pick_other_remote = len(rems) > 1
         other_remote_choice = "o[ther-remote]" if can_pick_other_remote else ""
         remote_branch = f"{new_remote}/{branch}"
         if not self.__git.get_commit_sha_by_revision(remote_branch, prefix="refs/remotes/"):
-            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'q', other_remote_choice))
+            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'Q', other_remote_choice))
             ask_message = f"Push untracked branch {bold(branch)} to {bold(new_remote)}?" + choices
             ask_opt_yes_message = f"Pushing untracked branch {bold(branch)} to {bold(new_remote)}..."
             ans = self.ask_if(ask_message, ask_opt_yes_message,
                               override_answer=None if self.__git.cli_opts.opt_push_untracked else "N")
-            if ans in ('y', 'yes', 'yq'):
-                self.__git.push(new_remote, branch)
-                if ans == 'yq':
-                    raise StopTraversal
-                self.flush_caches()
-            elif can_pick_other_remote and ans in ('o', 'other'):
-                self.__pick_remote(branch, is_called_from_traverse)
-            elif ans in ('q', 'quit'):
-                raise StopTraversal
-            return
+            if is_called_from_traverse:
+                if ans in ('y', 'yes', 'yq'):
+                    self.__git.push(new_remote, branch)
+                    if ans == 'yq':
+                        raise StopInteraction
+                    self.flush_caches()
+                elif can_pick_other_remote and ans in ('o', 'other'):
+                    self.__pick_remote(branch, is_called_from_traverse)
+                elif ans in ('q', 'quit'):
+                    raise StopInteraction
+                return
+            else:
+                if ans in ('y', 'yes'):
+                    self.__git.push(new_remote, branch)
+                    self.flush_caches()
+                    return
+                elif can_pick_other_remote and ans in ('o', 'other'):
```

Comment:
Whoops... actually, for the case of `can_pick_other_remote and ans in ('o', 'other'):`, should we `return` the entire method `handle_untracked_branch` after `__pick_remote` or not 🤔 it seems the behavior is different b/w `is_called_from_traverse` and `not is_called_from_traverse` cases now

Reply:
yes, there should be return